applications: Asset Tracker_v2: Split `prj.conf` into multiple overlays (#5662)
nrfconnect/sdk-nrf: applications/asset_tracker_v2/doc/asset_tracker_v2_description.rst

* :file:`prj.conf` - Configuration file common for all build targets
* :file:`boards/thingy91_nrf9160_ns.conf` - Configuration file specific for Thingy:91. This file is automatically merged with the :file:`prj.conf` file when you build for the ``thingy91_nrf9160_ns`` build target.
* :file:`boards/nrf9160dk_nrf9160_ns.conf` - Configuration file specific for nRF9160 DK. This file is automatically merged with the :file:`prj.conf` file when you build for the ``nrf9160dk_nrf9160_ns`` build target.
* :file:`overlay-nrf_cloud.conf` - Configuration file to set nRF Cloud as cloud provider.

> * :file:`overlay-aws.conf` - Configuration file to set AWS IoT as cloud provider.

Reviewer: ```suggestion
* :file:`overlay-aws.conf` - Configuration file to set AWS as cloud provider.
```

> * :file:`overlay-azure.conf` - Configuration file to set Azure IoT Hub as cloud provider.

Reviewer: ```suggestion
* :file:`overlay-azure.conf` - Configuration file to set Azure as cloud provider.
```

* :file:`overlay-agps.conf` - Configuration file to enable A-GPS.
* :file:`overlay-pgps.conf` - Configuration file to enable P-GPS.

> * :file:`overlay-low_power.conf` - Configuration file that achieves the lowest power consumption by disabling features that consume extra power like LED control and logging.

Reviewer 2: ```suggestion
* :file:`overlay-low_power.conf` - Configuration file that achieves the lowest power consumption by disabling features that consume extra power, such as LED control and logging.
```

Before building and running the firmware ensure that the cloud side is set up.
Also, the device must be provisioned and configured with the certificates according to the instructions for the respective cloud for the connection attempt to succeed.

The application defaults to using nRF Cloud as the cloud provider.

> However, this can be changed by specifying the overlay file that corresponds to another supported cloud provider when building the application.
> This is done by setting the CMake variable ``OVERLAY_CONFIG`` when calling the ``west build`` command.

Reviewer 2: ```suggestion
However, you can change the cloud provider by specifying the overlay file that corresponds to another supported cloud provider when building the application.
Set the CMake variable ``OVERLAY_CONFIG`` when calling the ``west build`` command.
```

> The ``OVERLAY_CONFIG`` CMake variable can be set to contain a list of overlays that will be patched in, each enabling a specific feature.

Reviewer 2: ```suggestion
You can set this variable to contain a list of overlays that will be patched in, each enabling a specific feature.
```

Before building and running the firmware ensure that the cloud side is set up.
Also, the device must be provisioned and configured with the certificates according to the instructions for the respective cloud for the connection attempt to succeed.

The application requires that a CMake list-variable `OVERLAY_CONFIG` is set when calling the west build command. `OVERLAY_CONFIG` contains a set

> of entries that enables certain features in the application by patching in the corresponding overlay configuration file. If the `OVERLAY_CONFIG` CMake variable is not set,

Reviewer: ```suggestion
of entries that enable certain features in the application by patching in the corresponding overlay configuration file. If the `OVERLAY_CONFIG` CMake variable is not set,
```

Reviewer 2: -> a set of entries that enables

> The application requires that a CMake list-variable `OVERLAY_CONFIG` is set when calling the west build command. `OVERLAY_CONFIG` contains a set
> of entries that enables certain features in the application by patching in the corresponding overlay configuration file. If the `OVERLAY_CONFIG` CMake variable is not set,

Reviewer 2: ```suggestion
The application requires that a CMake list-variable `OVERLAY_CONFIG` is set before calling the west build command.
The `OVERLAY_CONFIG` variable contains a set of entries that enable certain features in the application by patching in the corresponding overlay configuration file.
If the `OVERLAY_CONFIG` CMake variable is not set,
```

Reviewer 2: Remember, each sentence starts a new line.

> the application will default to using nRF Cloud as the cloud provider. See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the

Reviewer 2: ```suggestion
the application defaults to using nRF Cloud as the cloud provider.
See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the
```

> The application requires that a CMake list-variable `OVERLAY_CONFIG` is set when calling the west build command.
> The `OVERLAY_CONFIG` variable contains a set of entries that enable certain features in the application by patching in the corresponding overlay configuration file.
> If the `OVERLAY_CONFIG` CMake variable is not set, the application defaults to using nRF Cloud as the cloud provider.
> See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the same time.

Reviewer 2: ```suggestion
The application requires that you have set the CMake variable ``OVERLAY_CONFIG`` when calling the ``west build`` command.
The ``OVERLAY_CONFIG`` variable contains a set of entries that enable certain features in the application by patching in the corresponding overlay configuration file.
If the you have not set the ``OVERLAY_CONFIG`` CMake variable, the application defaults to using nRF Cloud as the cloud provider.
See :ref:`building_with_overlays` on how to combine overlay configuration files to enable multiple features at the same time.
```

.. external_antenna_note_start

.. note::

>    For the nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.

Reviewer 2: ```suggestion
   When you build the application for the nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` to achieve the best external antenna performance.
```

.. external_antenna_note_start

.. note::

>    For nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.

Reviewer 2: ```suggestion
   For the nRF9160 DK v0.15.0 and later, set the :kconfig:`CONFIG_GPS_MODULE_ANTENNA_EXTERNAL` option to ``y`` when building the application to achieve the best external antenna performance.
```

.. external_antenna_note_end

.. _building_with_overlays:

Building with overlays
======================

To build with Kconfig overlay, it must be based to the build system, as shown in the following example:

.. code-block:: console

   west build -b nrf9160dk_nrf9160_ns -- -DOVERLAY_CONFIG=overlay-low_power.conf

> The above command will build for nRF9160 DK using the configurations found in :file:`overlay-low_power.conf`, in addition to the configurations found in :file:`prj_nrf9160dk_nrf9160_ns.conf`.

Reviewer 2: ```suggestion
The above command builds for the nRF9160 DK using the configurations found in the :file:`overlay-low_power.conf` file, in addition to the configurations found in the :file:`prj_nrf9160dk_nrf9160_ns.conf` file.
```

   west build -b nrf9160dk_nrf9160_ns -- -DOVERLAY_CONFIG=overlay-low-power.conf

> The above command builds for the nRF9160 DK using the configurations found in the :file:`overlay-low-power.conf` file, in addition to the configurations found in the :file:`prj_nrf9160dk_nrf9160_ns.conf` file.

Reviewer 2: ```suggestion
This command builds for the nRF9160 DK using the configurations found in the :file:`overlay-low-power.conf` file, in addition to the configurations found in the :file:`prj_nrf9160dk_nrf9160_ns.conf` file.
```
